cognite/neat/_graph/loaders/_rdf2dms.py

```diff
@@ -425,70 +335,89 @@ def parse_direct_relation_to_unit(cls, value: Any, info: ValidationInfo) -> dict
+            return
 
-        return dm.NodeApply(
-            space=self.instance_space,
-            external_id=identifier,
-            type=(dm.DirectRelationReference(view_id.space, view_id.external_id) if type_ is not None else None),
-            sources=[
-                dm.NodeOrEdgeData(source=view_id, properties=dict(created.model_dump(exclude_unset=True).items()))
-            ],
-        )
-
-    def _create_edge_with_properties(
-        self,
-        identifier: str,
-        properties: dict[str | InstanceType, list[str]],
-        start_node: str,
-        end_node: str,
-        pydantic_cls: type[BaseModel],
-        view_id: dm.ViewId,
-    ) -> dm.EdgeApply:
-        type_ = properties.pop(RDF.type, [None])[0]
-        created = pydantic_cls.model_validate(properties)
-        if type_ is None:
-            raise ValueError(f"Missing type for edge {identifier}")
-
-        if self._unquote_external_ids:
-            identifier = urllib.parse.unquote(identifier)
+        if RDF.type not in properties:
+            error = ResourceCreationError(identifier, instance_type, "No rdf:type found")
+            if stop_on_exception:
+                raise error from None
+            yield error
+            return
+        type_ = properties.pop(RDF.type)[0]
+        if start_node and self._unquote_external_ids:
+            start_node = urllib.parse.unquote(start_node)
+        if end_node and self._unquote_external_ids:
+            end_node = urllib.parse.unquote(end_node)
+
+        if exclude:
+            properties = {k: v for k, v in properties.items() if k not in exclude}
+        if include:
+            properties = {k: v for k, v in properties.items() if k in include}
+        try:
+            sources = [
+                dm.NodeOrEdgeData(
+                    projection.view_id,
+                    projection.pydantic_cls.model_validate(properties).model_dump(exclude_unset=True),
+                )
+            ]
+        except ValueError as e:
+            error = ResourceCreationError(identifier, instance_type, f"Invalid properties: {e!s}")
+            if stop_on_exception:
+                raise error from None
+            yield error
+            return
 
-        return dm.EdgeApply(
-            space=self.instance_space,
-            external_id=identifier,
-            type=dm.DirectRelationReference(view_id.space, view_id.external_id),
-            start_node=dm.DirectRelationReference(self.instance_space, start_node),
-            end_node=dm.DirectRelationReference(self.instance_space, end_node),
-            sources=[
-                dm.NodeOrEdgeData(source=view_id, properties=dict(created.model_dump(exclude_unset=True).items()))
-            ],
-        )
+        if start_node and end_node:
+            yield dm.EdgeApply(
+                space=self.instance_space,
+                external_id=identifier,
+                type=(projection.view_id.space, type_),
```

Comment:
This one is tricky. It used to be `view_id.space, view_id.external_id` as the type, but that is wrong. This is also wrong. The type depends on the source instances, which we do not have access to her. Suggest solving this in a separate PR.